File: coursehub/src/main/java/com/coursehub/service/WishlistService.java
package com.coursehub.service;

import com.coursehub.dto.response.CourseResponse;

import java.util.List;
import java.util.UUID;

public interface WishlistService {
    void addToWishlist(UUID userId, UUID courseId);
    void removeFromWishlist(UUID userId, UUID courseId);
    boolean checkWishlist(UUID userId, UUID courseId);
    List<CourseResponse> getMyWishlist(UUID userId);
}

File: coursehub-frontend/src/services/wishlistService.ts
import axiosInstance from '../api/axiosInstance';
import type { ApiResponse } from '../types/auth.types';

export const wishlistService = {
  addToWishlist: async (courseId: string) => {
    const res = await axiosInstance.post<ApiResponse<void>>(`/wishlist/${courseId}`);
    return res.data;
  },

  removeFromWishlist: async (courseId: string) => {
    const res = await axiosInstance.delete<ApiResponse<void>>(`/wishlist/${courseId}`);
    return res.data;
  },

  checkWishlist: async (courseId: string) => {
    const res = await axiosInstance.get<ApiResponse<boolean>>(`/wishlist/${courseId}/check`);
    return res.data;
  },

  getMyWishlist: async () => {
    const res = await axiosInstance.get<ApiResponse<any[]>>('/wishlist');
    return res.data;
  },
};
